packages/tupaia-web/src/layout/ProjectCardList/LegacyProjectCard.tsx

```diff
@@ -105,13 +106,15 @@ const StyledPendingButton = styled(RouterButton).attrs({
 `;
 
 export const LegacyProjectDeniedLink = ({
-  url,
+  to,
   children,
+  routerState,
 }: {
-  url: string;
+  to: To;
   children: ReactNode;
+  routerState?: Record<string, unknown> | null;
 }) => (
-  <RouterButton to={url} color="primary" variant="outlined">
+  <RouterButton to={to} color="primary" variant="outlined" routerState={routerState}>
```

Comment:
There was a weird issue where if we used a URL `string` instead of a `To` object, the location updated multiple times and removed the router `state`, so I changed it to be a `To` object